packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -1184,10 +1183,14 @@ export class IndexedDbLruDelegate implements ReferenceDelegate, LruDelegate {
     txn: PersistenceTransaction,
     docKey: DocumentKey
   ): PersistencePromise<void> {
+    let totalBytesRemoved = 0;
+    const documentCache = this.db.getRemoteDocumentCache();
     return PersistencePromise.waitFor([
       documentTargetStore(txn).delete(sentinelKey(docKey)),
-      this.db.getRemoteDocumentCache().removeEntry(txn, docKey)
-    ]);
+      documentCache.removeEntry(txn, docKey).next(bytesRemoved => {
+        totalBytesRemoved += bytesRemoved;
+      })
+    ]).next(() => documentCache.updateSize(txn, -totalBytesRemoved));
```

Comment:
I remembered the other reason I didn't want to implement `removeEntries()`. For something like garbage collection, it's potentially a large number of entries, and I figured it would be better to remove them as we go rather than build up the array in memory. That being said, it would be an array of possibly thousands of document keys. Not the worst, if we decide that this API is too cumbersome as is.